forbidigo/forbidigo.go

```diff
@@ -200,19 +200,21 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 	// use that. It's used for matching unless usage of type information
 	// is enabled.
 	srcText := v.textFor(node)
-	matchText, pkgText := v.expandMatchText(node, srcText)
-	v.runConfig.DebugLog("%s: match %q, package %q", v.runConfig.Fset.Position(node.Pos()), matchText, pkgText)
+	matchTexts, pkgText := v.expandMatchText(node, srcText)
+	v.runConfig.DebugLog("%s: match %v, package %q", v.runConfig.Fset.Position(node.Pos()), matchTexts, pkgText)
 	for _, p := range v.linter.patterns {
-		if p.re.MatchString(matchText) &&
-			(p.Package == "" || p.pkgRe.MatchString(pkgText)) &&
-			!v.permit(node) {
-			v.issues = append(v.issues, UsedIssue{
-				identifier: srcText, // Always report the expression as it appears in the source code.
-				pattern:    p.re.String(),
-				pos:        node.Pos(),
-				position:   v.runConfig.Fset.Position(node.Pos()),
-				customMsg:  p.Msg,
-			})
+		for _, matchText := range matchTexts {
```

Comment:
If, for whatever reason, more than one match text is matched by the regex, then the same location is going to get reported more than once.

I'll merge your PR and then replace this loop with  a "regex matches any string" helper function.